src/ext/framework/mod.rs

```diff
@@ -254,90 +257,103 @@ impl Framework {
                     None => continue,
                 });
 
-                if let Some(command) = self.commands.get(&built) {
-                    if message.is_private() {
-                        if command.guild_only {
+                let groups = self.groups.clone();
```

Comment:
This can be a rather expensive operation per command dispatch. I can look into this personally once the PR lands